Adyen/Analytics/AnalyticsProvider/AnalyticsProvider.swift

```diff
@@ -86,6 +86,52 @@ internal final class AnalyticsProvider: AnalyticsProviderProtocol {
         }
     }
     
+    internal func add(info: AnalyticsEventInfo) {
+        eventDataSource.add(info: info)
+    }
+    
+    internal func add(log: AnalyticsEventLog) {
+        eventDataSource.add(log: log)
+        sendEventsIfNeeded()
+    }
+    
+    internal func add(error: AnalyticsEventError) {
+        eventDataSource.add(error: error)
+        sendEventsIfNeeded()
+    }
+    
+    internal func sendEventsIfNeeded() {
+        guard let request = requestWithAllEvents() else { return }
+        
+        apiClient.perform(request) { [weak self] result in
+            guard let self else { return }
+            // clear the current events on successful send
+            switch result {
+            case .success:
+                self.eventDataSource.removeAllEvents()
+                self.startNextTimer()
```

Comment:
Is this needed? The timer is repeating so it should just continue sending events?

Reply:
Thought it's better to reset the timer after a response since we are clearing the array here and if it triggers right after may be a futile effort.
Both is fine I guess?